### PR TITLE
CRSF DMA hint

### PR DESCRIPTION
**Pull request by andyp1per**
image

This also reduces the spam at startup

### CODE REVIEW

File: libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp [hunk 1]
```diff
@@ -158,14 +158,22 @@ void AP_CRSF_Telem::process_rf_mode_changes()
...
-    // warn the user if their setup is sub-optimal
-    if (_telem_last_report_ms == 0 && !uart->is_dma_enabled()) {
-        gcs().send_text(MAV_SEVERITY_WARNING, "CRSF: running on non-DMA serial port");
-    }
-    // report a change in RF mode or a chnage of more than 10Hz if we haven't done so in the last 5s
+    // report a change in RF mode or a change of more than 10Hz if we haven't done so in the last 5s
     if ((now - _telem_last_report_ms > 5000) &&
         (_telem_rf_mode != current_rf_mode || abs(int16_t(_telem_last_avg_rate) - int16_t(_scheduler.avg_packet_rate)) > 25)) {
         if (!rc().suppress_crsf_message()) {
+            // warn the user if their setup is sub-optimal
```
Comment (yaapu) [inline libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:165]: the way it is now users will get the message only when there's an RF mode change or rate change, is this intentional?

Comment (andyp1per) [reply libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:165]: I think we always get at least one mode change - currently I see a lot of spam at startup before the mode change has been detected

File: libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp [hunk 2]
```diff
@@ -158,14 +158,22 @@ void AP_CRSF_Telem::process_rf_mode_changes()
...
+            if (_telem_last_report_ms == 0 && !uart->is_dma_enabled()) {
```
Comment (Hwurzburg) [inline libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:166]: I think we may want to move this out from under the suppress message....its a one time thing, not repetitive "spam" that ELRS (or  quite frankly, I dont like seeing in my OSD all the time on CRSF either), but setting the surpress should not stop the one time warning....

Comment (andyp1per) [reply libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:166]: Ok, fixed

**Review by Hwurzburg — APPROVED** [review]

**Comment by tridge** [comment]
note that H7 doesn't need DMA for high baudrates

**Comment by andyp1per** [comment]
Thanks fixed

File: libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp [hunk 3]
```diff
@@ -167,13 +167,23 @@ void AP_CRSF_Telem::process_rf_mode_changes()
...
-    // warn the user if their setup is sub-optimal
-    if (_telem_last_report_ms == 0 && !uart->is_dma_enabled()) {
-        gcs().send_text(MAV_SEVERITY_WARNING, "CRSF: running on non-DMA serial port");
-    }
-    // report a change in RF mode or a chnage of more than 10Hz if we haven't done so in the last 5s
+    // report a change in RF mode or a change of more than 10Hz if we haven't done so in the last 5s
     if ((now - _telem_last_report_ms > 5000) &&
         (_telem_rf_mode != current_rf_mode || abs(int16_t(_telem_last_avg_rate) - int16_t(_scheduler.avg_packet_rate)) > 25)) {
+        // warn the user if their setup is sub-optimal
+#if !defined(STM32H7)
```
Comment (tridge) [inline libraries/AP_RCTelemetry/AP_CRSF_Telem.cpp:174]: i'd prefer just documentation and @SYS/UART.txt
@Hwurzburg we should discuss

**Review by tridge — CHANGES_REQUESTED** [review]

**Pull request closed without merging**